### ISSUE TITLE
Remove Migration creates invalid Modelsnapshot

### ISSUE THREAD

**Issue by JakobFerdinand** [aspnet/EntityFrameworkCore#13984]
Remove migration makes invalid changes to `ModelSnapshot` when using `TPH` with discriminator type `char`.

``` csharp
public abstract class Person
{
    public int Id { get; set; }
    public string Name { get; set; }
}
public class Customer : Person
{
    public decimal AmountSpent { get; set; }
}
public class Employee : Person
{
    public decimal Salary { get; set; }
}
```

``` csharp
public class SampleDbContext : DbContext
{
    public DbSet<Person> People { get; set; }
    public DbSet<Customer> Customers { get; set; }
    public DbSet<Employee> Employees { get; set; }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Entity<Person>(b =>
        {
            b.HasDiscriminator<char>("Type")
                .HasValue<Customer>('C')
                .HasValue<Employee>('E');
        });
    }
}
```

Important is the discriminator type `char`. The problem does not occur when using `int`, `short` or `string`. 

You can find a [sample Project here](https://github.com/rzlsoftware/EfCoreRemoveMigrationIssueSample).

#### Repro Steps
 - call Remove-Migration
 - look at changes in Model Snapshot
 - *expected **No Change***

![modelsnapshotdifference](https://user-images.githubusercontent.com/16666458/48765563-ab797e00-ecb2-11e8-8f8b-de3c9b73f561.PNG)

### Further technical details
EF Core version: 2.1.4
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10 18.03
IDE: Visual Studio 2017 15.9.2

**Comment by ajcvickers** [comment]
Note for triage: I was able to consistently reproduce this.

**Comment by ajcvickers** [comment]
Repros on 2.2

**Comment by bricelam** [comment]
Looks like this got fixed in 3.0.0
